nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -630,14 +631,49 @@ public static void checkTypeParameterNullnessForMethodOverriding(
    * String}.
    *
    * @param method the generic method
-   * @param enclosingType the enclosing type in which we want to know {@code method}'s return type
-   *     nullability
+   * @param enclosingSymbol the enclosing class in which we want to know {@code method}'s return
+   *     type nullability
    * @param state Visitor state
    * @param config The analysis config
    * @return nullability of the return type of {@code method} in the context of {@code
    *     enclosingType}
    */
   public static Nullness getGenericMethodReturnTypeNullness(
+      Symbol.MethodSymbol method, Symbol enclosingSymbol, VisitorState state, Config config) {
+    Type enclosingType = getTypeForSymbol(enclosingSymbol, state);
+    if (enclosingType == null) {
+      // we have no additional information from generics, so return NONNULL (presence of a @Nullable
+      // annotation should have been handled by the caller)
+      return Nullness.NONNULL;
+    }
+    return getGenericMethodReturnTypeNullness(method, enclosingType, state, config);
+  }
+
+  /**
+   * Get the type for the symbol, accounting for anonymous classes
+   *
+   * @param symbol the symbol
+   * @param state the visitor state
+   * @return the type for {@code symbol}
+   */
+  @Nullable
+  private static Type getTypeForSymbol(Symbol symbol, VisitorState state) {
+    if (symbol.isAnonymous()) {
+      // For anonymous classes, symbol.type does not contain annotations on generic type parameters.
+      // So, we get a correct type from the enclosing NewClassTree.
+      TreePath path = state.getPath();
+      NewClassTree newClassTree = ASTHelpers.findEnclosingNode(path, NewClassTree.class);
+      if (newClassTree == null) {
+        throw new RuntimeException(
+            "method should be inside a NewClassTree " + state.getSourceForNode(path.getLeaf()));
+      }
```

Comment:
Do we want a check that the `NewClassTree` is for the same type as `symbol`['s supertype]? Or can that be safely assumed to never break?

Reply:
Can we add a test case for when the class name is given as something other than a simple name? `new Foo.Builder<@Nullable String>() { ... }` or even `new Foo<@Nullable String>.Builder() {...}`?

Reply:
> Do we want a check that the `NewClassTree` is for the same type as `symbol`['s supertype]? Or can that be safely assumed to never break?

I added a check in af6ca1fb5da1a0e9b6e4db4e3d805e6dd7d0bb70; can't hurt :-)

> Can we add a test case for when the class name is given as something other than a simple name? `new Foo.Builder<@Nullable String>() { ... }` or even `new Foo<@Nullable String>.Builder() {...}`?

I added some tests in 16eb9c2f6f531939617dbf3f765a1e8d6557375e but one of them is failing 😐 I propose I fix that one in a follow-up; does that sound ok?